### DISCUSSION TITLE
How to implemenet EdgeConv for hyperedges?

### DISCUSSION THREAD

**Discussion by Shihab-Shahriar** [pyg-team/pytorch_geometric#9013] [Q&A]
Hi, I have a 3-node hyperedges in my graph, and I'd like to extend the EdgeConv to work with that, something like-

```
def message(self, x_i, x_j, x_k):
    input = torch.cat([x_i-x_j, x_i-x_k])
```

Is it possible to do this using PyG?

**Comment by wsad1** [comment]
`EdgeConv` isn't really built for hyper graphs. Pyg has [HyperGraphConv](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.HypergraphConv.html#torch_geometric.nn.conv.HypergraphConv). You could maybe build a version of `EdgeConv` that supports hypergraphs. Hope this helps.
